MyPerfectOnboardingApplication/Sources/MyPerfectOnboarding.Database/Repository/ListRepository.cs

```diff
@@ -1,39 +1,43 @@
-﻿using System;
-using System.Threading.Tasks;
+﻿using MongoDB.Driver;
 using MyPerfectOnboarding.Contracts.Database;
 using MyPerfectOnboarding.Contracts.Models;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MyPerfectOnboarding.Database.Models;
 
 namespace MyPerfectOnboarding.Database.Repository
 {
     internal class ListRepository : IListRepository
     {
-        private readonly ListItem[] _items = {
-            new ListItem
-            {
-                Id = new Guid("11AC59B7-9517-4EDD-9DDD-EB418A7C1644"),
-                Text = "aaaaa",
-                IsActive = false,
-                CreationTime = new DateTime(1589, 12, 3),
-                LastUpdateTime = new DateTime(1896, 4, 7)
-            },
-            new ListItem
-            {
-                Id = new Guid("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAC"),
-                Text = "dfads",
-                IsActive = false,
-                CreationTime = new DateTime(4568, 6, 23),
-                LastUpdateTime = new DateTime(8569, 8, 24)
-            },
-        };
-
-        public async Task<ListItem> AddItemAsync(ListItem item) => await Task.FromResult(_items[1]);
-
-        public async Task<ListItem> DeleteItemAsync(Guid id) => await Task.FromResult(_items[0]);
-
-        public async Task ReplaceItemAsync(ListItem editedItem) => await Task.CompletedTask;
-
-        public async Task<ListItem[]> GetAllItemsAsync() => await Task.FromResult(_items);
-
-        public async Task<ListItem> GetItemAsync(Guid itemId) => await Task.FromResult(_items[0]);
+        private readonly IMongoCollection<IListItem> _collection;
+
+        private const string NameOfDBCollection = "items";
+
+        public ListRepository(IConnectionDetails connectionDetails)
+        {
+            var url = MongoUrl.Create(connectionDetails.DataConnectionString);
+            var client = new MongoClient(url);
+            var db = client.GetDatabase(url.DatabaseName);
+            _collection = db.GetCollection<IListItem>(NameOfDBCollection);
+        }
+
+        public async Task<ListItem> AddItemAsync(ListItem item)
+        {
+            await _collection.InsertOneAsync(new ListItemModel(item));
+            return item;
+        }
+
+        public async Task<ListItem> DeleteItemAsync(Guid id)
+            => await _collection.FindOneAndDeleteAsync(item => item.Id == id).AsImmutable();
+
+        public async Task<ListItem> ReplaceItemAsync(ListItem editedItem)
+            => await _collection.FindOneAndReplaceAsync(item => item.Id == editedItem.Id, new ListItemModel(editedItem)).AsImmutable();
```

Comment:
These oneliners are tricky. They are easy to grow off the screen and hard to follow IMO. Consider rewrapping all body-less methods in this class.

Unfortunately, R# does not help you much here, you can set `Wrap chained method calls` but the only thing it does is it makes wrapping look awful (personal opinion) and thus enforces you to reformat it by hand 😆 

This very method would IMO be a bit more readable when formatted like this:
```suggestion
            => await _collection
                .FindOneAndReplaceAsync(
                    item => item.Id == editedItem.Id,
                    new ListItemModel(editedItem))
                .AsImmutable();
```